feat(discover2) Move routes and add opt-in/out buttons

markstory opened the pull request:
With this change users who have access to both discover variants will now only see discover2 in their sidebar. However, they will also have opt-out buttons at the bottom of the discover2 view. This state is stored in localStorage as we plan to have a fairly short overlap period (a few weeks) and adding an organization wide setting seemed arduous.

I've also moved the URLs for discover2 to be closer to their final resting place. The `/discover/queries` URL is not final. We'd like to use `/discover` for the query list but that route is currently occupied by discover1 and I didn't want to complect both discovers with more conditional code. Once discover1 has been removed we can move the canonical route for discover2 saved queries to `/discover` and keep `/discover/queries` as an alias.

Reload events added in getsentry/reload#135

Switching buttons:
<img width="355" alt="Screen Shot 2020-01-22 at 12 43 38 PM" src="https://user-images.githubusercontent.com/24086/72919119-e642d000-3d14-11ea-934c-8c2226cf1ca2.png">

<img width="325" alt="Screen Shot 2020-01-22 at 8 55 41 AM" src="https://user-images.githubusercontent.com/24086/72918819-53099a80-3d14-11ea-85d4-a38950bc1d1c.png">

markstory committed:
ref(discover2) Make discover 2 routes use the discover URL prefix

This causes some additional components to remount on each URL transition
but I don't know of another way to have two different component trees
share a single URL prefix.

markstory committed:
feat(discover) Add discover toggle for users with access to both.

With this change users who have access to both discover variants will
now only see discover2 in their sidebar. However, they will also have
opt-out buttons at the bottom of the discover2 view. This state is
stored in localStorage as we plan to have a fairly short overlap period
(a few weeks) and adding an organization wide setting seemed arduous.

I've also moved the URLs for discover2 to be closer to their final
resting place. The `/discover/queries` URL is not final. We'd like to
use `/discover` for the query list but that route is currently occupied
by discover1 and I didn't want to complect both discovers with more
conditional code. Once discover1 has been removed we can move the
canonical route for discover2 saved queries to `/discover` and keep
`/discover/queries` as an alias.

markstory committed:
Fix alignment

markstory committed:
Fix paths in acceptance tests.

markstory committed:
Discover should retain globalselection header state.

markstory committed:
Merge branch 'master' into ref/discover-routes

markstory committed:
Remove more dead code.

markstory committed:
Fix another URL

markstory committed:
Use getResultsViewUrlTarget() in more places.

dashed commented on src/sentry/static/sentry/app/components/sidebar/index.jsx:
```diff
@@ -285,21 +296,23 @@ class Sidebar extends React.Component {
-                  </Feature>
+                  {discoverVersion === '2' && (
+                    <Feature features={['events-v2']} organization={organization}>
+                      <SidebarItem
```
@markstory I think we can set `active` prop to `true` on desired matching routes from React Router for the Query builder and the event details page.

markstory replied:
Found a different solution. Icon highlighting should be good now.

markstory committed:
Merge branch 'master' into ref/discover-routes

markstory committed:
Hide events in the sidebar and fix icon highlights

Events + discover1 should be hidden when discover2 is active. Add a hack
for discover as the URLs are annoying right now.

dashed reviewed (approved):
Looks good to me 👍

markstory committed:
Merge branch 'master' into ref/discover-routes

markstory committed:
Merge branch 'master' into ref/discover-routes

markstory committed:
Fix defaulting of discover/events switches.
